app/services/katello/pulp3/repository/apt.rb

```diff
@@ -40,7 +40,7 @@ def sanitize_pulp_distribution(distribution)
         def pulp_distributions
           return [] if repo.version_href.blank?
           return ["default"] if version_missing_structure_content?
-          pulp_primary_api.content_release_components_api.list({:repository_version => repo.version_href}).results.map { |x| sanitize_pulp_distribution(x.distribution) }.uniq
+          pulp_primary_api.content_release_components_api.list({:repository_version => repo.version_href}).results.map { |x| sanitize_pulp_distribution(x.distribution) }.uniq.sort
```

Comment:
This would make it so that a given set of distributions, must always be in a fixed order going forward. This makes comparisons whether something has changed since last time easier. Especially since we don't save these parameters as a list, but often dump things in a URL options string formatted as `"/?comp=#{components}&rel=#{distributions}"`. Deconstructing this back into lists for order independent list comparisons is quite tedious.

I think we could live with the fact that this only takes effect "going forward", since a one time extra update content action per repository is not actually that expensive (they happen all the time anyway), and the next one would bring things into the standardized order.